generator/writers.ts

```diff
@@ -0,0 +1,92 @@
+import * as handlebars from 'handlebars';
+import * as path from 'path';
+
+import {loadContributorFromConfiguration} from './contributors.js';
+import {GuideHTMLFileWithMetadata, LearningPath, PathTopic, RootCards, SerializedGuideJson, TopLevelFile} from './file-types.js';
+import * as fs from './fsp.js';
+import {markdown} from './markdown.js';
+
+const TEMPLATE_DIRECTORY = path.resolve(__dirname, '..', '..', 'templates');
+
+function readTemplate(templateName: string) {
+  return handlebars.compile(fs.readFileSync(
+      path.resolve(TEMPLATE_DIRECTORY, `${templateName}.html`), 'utf8'));
+}
+
+const PATH_TEMPLATE = readTemplate('path');
+const DEVSITE_TEMPLATE = readTemplate('devsite');
+const ROOT_CARDS_TEMPLATE = readTemplate('root-cards');
+const GUIDE_TEMPLATE = readTemplate('guide');
+const CODELAB_TEMPLATE = readTemplate('glitch');
+
+export async function writeTopLevelFile(directory: string, file: TopLevelFile) {
+  await fs.writeFile(path.resolve(directory, file.name), file.body);
+}
+
+async function writeSingleGuide(
+    directory: string, learningPath: LearningPath, topic: PathTopic,
+    guide: GuideHTMLFileWithMetadata) {
```

Comment:
I was wondering why this type name includes `HTML`. Isn't it a JS object containing markdown?

Reply:
It is, but it is based on the HTML file that we read. Mostly for me to keep track of what the original value was. In general, naming is bonkers atm and I should fix that first.